### PR TITLE
Reduce amount of function pointers in query invocation.

### PR DESCRIPTION
**Pull request by cjgillot**
r? @ghost

### CODE REVIEW

**Comment by bjorn3** [comment]
Mostly improvements up to 0.5%, but also a couple of regressions up to 1.6%.

File: compiler/rustc_query_impl/src/plumbing.rs [hunk 1]
```diff
@@ -389,14 +389,15 @@ macro_rules! define_queries {
...
 
             #[inline]
-            fn compute(tcx: QueryCtxt<'tcx>, key: Self::Key) -> Self::Value {
-                let is_local = key.query_crate() == LOCAL_CRATE;
-                let provider = if is_local {
+            fn compute_fn(tcx: QueryCtxt<'tcx>, key: &Self::Key) ->
+                fn(TyCtxt<'tcx>, Self::Key) -> Self::Value
+            {
+                let is_local = key.query_crate_is_local();
+                if is_local {
```
Comment (bjorn3) [inline compiler/rustc_query_impl/src/plumbing.rs:396]:
```suggestion
                if key.query_crate_is_local() {
```

File: compiler/rustc_query_impl/src/keys.rs [hunk 2]
```diff
@@ -42,8 +44,9 @@ impl<'tcx> Key for ty::Instance<'tcx> {
 }
 
 impl<'tcx> Key for mir::interpret::GlobalId<'tcx> {
-    fn query_crate(&self) -> CrateNum {
-        self.instance.query_crate()
+    #[inline(always)]
+    fn query_crate_is_local(&self) -> bool {
+        self.instance.query_crate_is_local()
```
Comment (bjorn3) [inline compiler/rustc_query_impl/src/keys.rs:49]:
```suggestion
        true
```

**Review by bjorn3 — APPROVED** [review]
Overall improvements up to 0.6% with only a few small regressions up to 0.3%. No impact on bootstrap time.

**Comment by JohnTitor** [comment]
Looks like it's ready to review?

**Comment by cjgillot** [comment]
Nobody is assigned yet.
r? @bjorn3 (please reassign if you can't)

**Pull request merged**